linera-chain/src/chain.rs

```diff
@@ -72,6 +72,37 @@ pub struct ChainTipState {
     pub next_block_height: BlockHeight,
 }
 
+impl ChainTipState {
+    /// Checks that the proposed block is suitable, i.e. at the expected height and with the
+    /// expected parent.
+    pub fn check_proposed_block(&self, new_block: &Block) -> Result<(), ChainError> {
```

Comment:
This name is a bit confusing because we re-use it for something else (safety rules) in the manager.
Maybe `verify_block_chaining` ?